### DISCUSSION TITLE
Is it possible to use RandShiftIntensity to apply to a specific area in an image?

### DISCUSSION THREAD

**Discussion by Minxiangliu** [Project-MONAI/MONAI#4028] [Q&A] [answered]
`RandShiftIntensity` is one of the good data augmentation methods, now I have images with `0 values` in the background, and then I only want `Shift` on the foreground (not the background), is this possible? Or can anyones suggest me how to do it? 

Thanks in advance!

image:
![image](https://user-images.githubusercontent.com/30144428/160975112-bbb54cd9-337a-46a6-a856-849c9fa55cad.png)

**Comment by yiheng-wang-nv** [comment, marked as answer]
Hi @Minxiangliu , `RandShiftIntensity` cannot support your requirement directly. You may need to define a customized transform. Here is an example:
```
from monai.utils.enums import TransformBackends
from monai.transforms.transform import Transform
from monai.transforms import RandShiftIntensity
import torch

class FgShift(Transform):

    backend = [TransformBackends.TORCH]

    def __init__(self, offsets, prob = 0.1, bg_value = 0.0):
        self.shifter = RandShiftIntensity(offsets=offsets, prob=prob)
        self.bg_value = bg_value

    def __call__(self, img, offset = None):
        """
        Apply the transform to `img`.
        """
        out = torch.where(img != self.bg_value, self.shifter(img), img)

        return out
```
The following is a simple test code to show the result:
```
trans = FgShift(prob=1.0, offsets=(1, 4), bg_value=0.0)
a = torch.tensor([[-1.0, 2.5, 3.0], [1.1, 0, 0]])

for i in range(5):
    print(f"output is: {trans(a)} offset is: {trans.shifter._offset}")
    print(" ")
```

**Reply by Minxiangliu** [reply]
Thank you for your concise explanation, I can understand what you mean. In addition, can the method of `monai.transforms.RandScaleIntensity` be similar to your approach? This is because my background is 0, so multiplying by any number won't change.

**Reply by yiheng-wang-nv** [reply]
Hi @Minxiangliu , since your bg_value is 0, you can use `RandScaleIntensity` directly and do not need to wrap it into a new transform.
